core/chainio/avs_subscriber.go

```diff
@@ -147,6 +154,90 @@ func subscribeToNewTasks(
 	return nil, fmt.Errorf("Failed to subscribe to new AlignedLayer tasks after %d retries", MaxRetries)
 }
 
+func (s *AvsSubscriber) processNewBatch(batch *servicemanager.ContractAlignedLayerServiceManagerNewBatch, batchesSet map[[32]byte]struct{}, newBatchMutex *sync.Mutex, newTaskCreatedChan chan<- *servicemanager.ContractAlignedLayerServiceManagerNewBatch) {
+	newBatchMutex.Lock()
+	defer newBatchMutex.Unlock()
+
+	if _, ok := batchesSet[batch.BatchMerkleRoot]; !ok {
+		s.logger.Info("Received new task", "batchMerkleRoot", hex.EncodeToString(batch.BatchMerkleRoot[:]))
+		batchesSet[batch.BatchMerkleRoot] = struct{}{}
+		newTaskCreatedChan <- batch
+
+		// Remove the batch from the set after RemoveBatchFromSetInterval time
+		go func() {
+			time.Sleep(RemoveBatchFromSetInterval)
+			newBatchMutex.Lock()
+			delete(batchesSet, batch.BatchMerkleRoot)
+			newBatchMutex.Unlock()
+		}()
+	}
+}
+
+// getLatestTaskFromBlockchain queries the blockchain for the latest task using the FilterLogs method.
+// The alternative to this is using the FilterNewBatch method from the contract's filterer, but it requires
+// to iterate over all the logs, which is not efficient and not needed since we only need the latest task.
+func (s *AvsSubscriber) getLatestTaskFromBlockchain() (*servicemanager.ContractAlignedLayerServiceManagerNewBatch, error) {
```

Comment:
```suggestion
func (s *AvsSubscriber) getLatestTaskFromEthereum() (*servicemanager.ContractAlignedLayerServiceManagerNewBatch, error) {
```

Reply:
Solved in [2312232](https://github.com/yetanotherco/aligned_layer/pull/703/commits/2312232befa6ac0f87ff49b614d1cebd3fb0009d)